### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -5,8 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+    let separ = separator;
+
+    if (separator === undefined) {
+      separ = ',';
+    }
```
Comment (EnjSadman) [inline src/arrayMethodJoin.js:13]:
```suggestion
    [].__proto__.join2 = function(separator = ',') {
```
can usr this instead of if

**Review by EnjSadman — APPROVED** [review]

File: src/arrayMethodJoin.js (hunk 1 above)
Comment (Anna-Tkachenko) [reply src/arrayMethodJoin.js:13]: Agree, set default value for separator in function declaration

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -5,8 +5,29 @@
...
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        res += separ;
+      }
```
Comment (Anna-Tkachenko) [inline src/arrayMethodJoin.js:18]: Use key word continue here to go to the next iteration and avoid the next if / else statement

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/arrayMethodJoin.js (hunk 2 above)
Comment (yelnikov-andrii) [reply src/arrayMethodJoin.js:18]: it doesn't work if I use continue, because it needs to add a separator

Comment (Anna-Tkachenko) [reply src/arrayMethodJoin.js:18]: Add continue after the string with adding separator

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
You didn't fix comments

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**